DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/L1/test_l1_port_state.py

```diff
@@ -0,0 +1,88 @@
+    pytest.mark.usefixtures("cleanup_tgen")
+]
+
+async def assets(rc, out, timeout):
+    assert rc == 0, f"Failed to get a grep count of entities.\n"
+    err_msg = f"FAIL: Not all ports exist and time is up, expected time <{timeout}.\n"
+    assert out.strip() == "1", err_msg
+
+async def test_l1_port_state(testbed):
+    """
+    Test Name: test_l1_port_state
+    Test Suite: suite_functional_l1
+    Test Overview: Verify port link and timings status.
+    Test Author: Kostiantyn Stavruk
+    Test Procedure:
+    1.  Set entities swp1, swp2, swp3, swp4 UP state.
+    2.  Verify the time it took for all ports to get up.
+
+    """
+
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 4)
+    if not tgen_dev or not dent_devices:
+        print("The testbed does not have enough dent with tgen connections")
+        return
+    dent_dev = dent_devices[0]
+    device_host_name = dent_dev.host_name
+    tg_ports = tgen_dev.links_dict[device_host_name][0]
+    ports = tgen_dev.links_dict[device_host_name][1]
+    timeout = 30
+
+    address_map = (
+        # swp port, tg port,    tg ip,     gw,        plen
+        (ports[0], tg_ports[0], "1.1.1.2", "1.1.1.1", 24),
+        (ports[1], tg_ports[1], "1.1.1.3", "1.1.1.1", 24),
+        (ports[2], tg_ports[2], "1.1.1.4", "1.1.1.1", 24),
+        (ports[3], tg_ports[3], "1.1.1.5", "1.1.1.1", 24),
+    )
+
+    dev_groups = tgen_utils_dev_groups_from_config(
+        {"ixp": port, "ip": ip, "gw": gw, "plen": plen}
+        for _, port, ip, gw, plen in address_map
+    )
+
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, ports, dev_groups)
+
+    for port in ports:
+        rc, out = await dent_dev.run_cmd(f"ifconfig -a | grep -w '{port}' | wc -l")
+        if rc != 0 or out.strip() != "1":
+            time.sleep(timeout)
+            rc, out = await dent_dev.run_cmd(f"ifconfig -a | grep -w '{port}' | wc -l")
+            await assets(rc, out, timeout)
+        else:
+            await assets(rc, out, timeout)
+
+    out = await IpLink.set(
+        input_data=[{device_host_name: [
+            {"device": port, "operstate": "up"} for port in ports]}])
+    assert out[0][device_host_name]["rc"] == 0, f"Verify that entities set to 'UP' state.\n{out}"
+
+    for _ in range(20):
+        out = await IpLink.show(input_data=[{device_host_name: [{"cmd_options": "-j"}]}],
+                                parse_output=True)
+        assert out[0][device_host_name]["rc"] == 0, f"Failed to get links.\n"
+
+        links = out[0][device_host_name]["parsed_output"]
+        link_is_up = []
+        for port in ports:
+            for link in links:
+                    if port in link["ifname"]:
+                        link_is_up.append(link["operstate"] == "UP")
+                        break
+        if all(link_is_up):
+            break
+        time.sleep(timeout/6)
+    else:
+        raise Exception(f"Not even single swp is UP.\n")
```

Comment:
Better make it as assertion:
assert all(link is up), "One of the ports is not in the UP state"